### PR TITLE
DRILL-7008: Drillbits: clear stale shutdown hooks

### PR DESCRIPTION
**Pull request by vlsi**
`ShutdownThread` is no longer required when `Drillbit#close()` is called.

mvn install for Drill project consumed 600MiB (there were 160 shutdown hooks)

### CODE REVIEW

**Comment by vlsi** [comment]
By the way, it improves Travis build time from 42 to 33 minutes:

**Comment by arina-ielchiieva** [comment]
@vlsi thanks for the PR. Could you please create Apache Jira and add it's number in commit message and PR title?

**Comment by vlsi** [comment]
@arina-ielchiieva , done

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/Drillbit.java [hunk 1]
```diff
@@ -258,6 +260,13 @@ public synchronized void close() {
     }
     final Stopwatch w = Stopwatch.createStarted();
     logger.debug("Shutdown begun.");
+    // We don't really want for Drillbits to pile up in memory, so the hook should be removed
+    // It might be better to use PhantomReferences to cleanup as soon as Drillbit becomes
+    // unreachable, however current approach seems to be good enough.
+    Thread shutdownHook = this.shutdownHook;
+    if (shutdownHook != null && Thread.currentThread() != shutdownHook) {
+      Runtime.getRuntime().removeShutdownHook(shutdownHook);
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/Drillbit.java]: @vlsi `removeShutdownHook` can throw `IllegalStateException` if the virtual machine is already in the process of shutting down. Maybe we should add try / catch block to ignore such exception?

Comment (vlsi) [reply exec/java-exec/src/main/java/org/apache/drill/exec/server/Drillbit.java]: \@arina-ielchiieva , it is handled by  `Thread.currentThread() != shutdownHook`, isn't it?

Comment (vlsi) [reply exec/java-exec/src/main/java/org/apache/drill/exec/server/Drillbit.java]: @arina-ielchiieva , I've added try-catch

Comment (arina-ielchiieva) [reply exec/java-exec/src/main/java/org/apache/drill/exec/server/Drillbit.java]: Thanks.

**Comment by arina-ielchiieva** [comment]
+1, @vlsi thanks for making the changes and the fact that tests run much faster is awesome.

**Pull request closed without merging**